vm/src/builtins/bytearray.rs

```diff
@@ -373,9 +375,12 @@ impl PyByteArray {
         self.inner().hex(sep, bytes_per_sep, vm)
     }
 
-    #[pymethod]
-    fn fromhex(string: PyStrRef, vm: &VirtualMachine) -> PyResult<PyByteArray> {
-        Ok(PyBytesInner::fromhex(string.as_str(), vm)?.into())
+    #[pyclassmethod]
+    fn fromhex(cls: PyTypeRef, string: PyStrRef, vm: &VirtualMachine) -> PyResult {
+        let bytes = PyBytesInner::fromhex(string.as_str(), vm)?;
+        let bytes = vm.ctx.new_bytes(bytes);
```

Comment:
to check if this is intended: this is `bytearray` but calling `new_bytes`, right?

Reply:
yes, see the updated test code. the bytes will pass to __new__ or __init__ in python than back to tp_new.